### PR TITLE
[#188] Push image to a Docker registry

### LINKED ISSUE #188: Automatically push image to Docker registry
Pushing Docker images directly from the CI in response to tag events, would be helpful in order to keep the image up to date

### PR DESCRIPTION
**Pull request by sestrella**
Fixes #188

### CODE REVIEW

File: .github/workflows/docker.yml
Comment (CristhianMotoche) [inline .github/workflows/docker.yml:31]: @sestrella I don't think we'll push a tags to create new releases very frequently. Out of curiosity, how long is the cache available to be used in GH Actions?

File: .github/workflows/docker.yml
Comment (CristhianMotoche) [inline .github/workflows/docker.yml:53]: @sestrella Oh, I see our plan is to build on every commit push and push the image when the ref is a tag. That makes sense. Although, I think we should also avoid the step `docker/login-action@v1` on every push. I guess the builds will fail when users out of the organization do a contribution since they're not part of our GH teams.

**Review by CristhianMotoche — COMMENTED** [review]
@sestrella LGTM! I left a question to know if external contributors will get blocked in the docker build step. Please, let me know your thoughts.

File: .github/workflows/docker.yml
Comment (sestrella) [reply .github/workflows/docker.yml:53]: @CristhianMotoche not really, since pushing tags requires write privileges to the repo, therefore, it can’t only be done by admins and maintainers, I don't think this is going to be an issue for external contributors.

Comment (sestrella) [reply .github/workflows/docker.yml:53]: Good catch, about `docker/login-action@v1` I'm going to update that

Comment (CristhianMotoche) [reply .github/workflows/docker.yml:53]: Hey @sestrella I think my previous comment was a bit confusing. I was referring to push commits from external contributors since I see that the `docker/login-action@v1` step is executed on every commit push. Then, it could fail for external contributors, couldn't it?

File: .github/workflows/docker.yml
Comment (sestrella) [reply .github/workflows/docker.yml:31]: @CristhianMotoche here are some insights:

**Without cache:**

image

**With cache:**

image

Comment (CristhianMotoche) [reply .github/workflows/docker.yml:31]: Ignore this comment, too.

File: .github/workflows/docker.yml
Comment (sestrella) [reply .github/workflows/docker.yml:7]: The idea is to build the image always to catch errors and only pushing on tag creation events.

File: .github/workflows/docker.yml
Comment (CristhianMotoche) [inline .github/workflows/docker.yml:81]: @sestrella I see some duplicated steps between `build` and `publish`. Is it possible to use [node anchors](https://en.wikipedia.org/wiki/YAML#Advanced_components) and references (or [merge](https://yaml.org/type/merge.html)) to make it DRY?

Comment (sestrella) [reply .github/workflows/docker.yml:81]: @CristhianMotoche unfortunately, YAML anchors are [not supported](https://github.community/t/support-for-yaml-anchors/16128) yet, on the other hand, there is something called [composite run steps](https://github.blog/changelog/2020-08-07-github-actions-composite-run-steps/) but it involves setting up a new repo which adds a level of indirection, so I think some code duplication in the context of this workflow is OK. What do you think?

Comment (CristhianMotoche) [reply .github/workflows/docker.yml:81]: Too bad anchors are not supported yet. I don't think we should complicate things with the _composite run steps_. Let's keep it simple.

File: .github/workflows/docker.yml
Comment (CristhianMotoche) [inline .github/workflows/docker.yml:46]: @sestrella Is it possible to add something like this for the `build` step but referring to `build` job from the `ci` workflow? In that way, if the CI workflow fails then this workflow won't run.

**Review by CristhianMotoche — APPROVED** [review]
@sestrella LGTM! I left a last question. Although, I don't think it's a blocker. Thanks for the contribution!

**Pull request merged**
